### PR TITLE
SCRUM-3125: support symbol edition in allele details page

### PR DESCRIPTION
**Pull request by abecerra**

### CODE REVIEW

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    dispatch({ 
+      type: 'EDIT_OBJECT', 
+      entityType: 'alleleSymbol', 
+      field: "internal", 
+      value: event.target?.value?.name
+    });
+  };
+
+  const synonymScopeOnChangeHandler = (props, event) => {
+    dispatch({
```
Comment (markquintontulloch) [inline src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js]: Need the following two lines before the dispatch function:
`//temporary solution -- replace with props.editorCallback() after PrimeReact upgrade 
    props.rowData.synonymScope = event.target.value;`
Also need to do the equivalent for internalOnChangeHandler and nameTypeOnChangeHandler.

**Review by markquintontulloch — CHANGES_REQUESTED** [review]
Looks good, just need to add the workaround for the bug we found earlier in the week as mentioned in my comment

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import { Button } from "primereact/button";
+import { FormTableWrapper } from "../../../components/FormTableWrapper";
+import { useRef } from "react";
+import { SymbolFormTable } from "./SymbolFormTable";
+
+export const SymbolFormEditor = ({ labelColumnSize, state, dispatch }) => {
```
Comment (markquintontulloch) [inline src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js]: Can we rename this (and the file) to SymbolForm for consistency with the other tables on the detail page?

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js (hunk 1 above)
Comment (abecerra) [reply src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js]: applied

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js (hunk 2 above)
Comment (abecerra) [reply src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js]: renamed

Comment (markquintontulloch) [reply src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js]: @abecerra Looks like you didn't commit the renamed file?

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolForm.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      type: 'EDIT_OBJECT', 
+      entityType: 'alleleSymbol', 
+      field: "nameType", 
+      value: event.target.value
+    });
+  };
+
+  const internalOnChangeHandler = (props, event) => {
+     // TODO -- replace with props.editorCallback() after PrimeReact upgrade
+     props.rowData.synonymScope = event.target.value;
```
Comment (markquintontulloch) [inline src/main/cliapp/src/containers/allelesPage/symbols/SymbolForm.js]: needs to be props.rowData.internal

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolForm.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    })
+  };
+
+  const onRowEditChange = (e) => {
+    return null;
+  };
+
+  const nameTypeOnChangeHandler = (props, event) => {
+     // TODO -- replace with props.editorCallback() after PrimeReact upgrade
+     props.rowData.synonymScope = event.target.value;
```
Comment (markquintontulloch) [inline src/main/cliapp/src/containers/allelesPage/symbols/SymbolForm.js]: needs to be props.rowData.nameType

File: src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js (hunk 2 above)
Comment (abecerra) [reply src/main/cliapp/src/containers/allelesPage/symbols/SymbolFormEditor.js]: It should be there, I had to rebase the branch with alpha, I'll fix the  props  values!

**Review by markquintontulloch — APPROVED** [review]

**Pull request merged**